File: subjects/2023-2024-RO_VARIANT_I/alg_prog/CSharp/CSharp/encoder/NaiveEncoding.cs
namespace CSharp.encoder;

public class NaiveEncoding : Encoder
{

    public override string encode(string m)
    {
        string result = "";

        for (var i = m.Length - 1; i >= 0; i--)
        {
            result += m[i];
        }

        return result;
    }
}

File: subjects/2023-2024-RO_VARIANT_I/alg_prog/Java/src/encoder/NaiveEncoding.java
package encoder;

public class NaiveEncoding extends Encoder {
    @Override
    public String encode(String m) {
        StringBuilder result = new StringBuilder();

        for (int i = m.length() - 1; i >= 0; i--) {
            result.append(m.charAt(i));
        }

        return result.toString();
    }
}
